src/digraph.jl

```diff
@@ -18,6 +18,26 @@ end
 
 DiGraph() = DiGraph(0)
 
+function DiGraph{T<:Real}(adjmx::SparseMatrixCSC{T})
+    dima, dimb = size(adjmx)
+    if dima != dimb
+        error("Adjacency / distance matrices must be square")
+    else
+        g = DiGraph(dima)
+        maxc = length(adjmx.colptr)
```

Comment:
It might be dimb+1, actually:

```
julia> a = spzeros(100000,100000)
100000x100000 sparse matrix with 0 Float64 entries:

julia> length(a.colptr)
100001
```